### PR TITLE
Enable defining new rho based RSpace benches

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
This pr enables benchmarking arbitrary contracts with jmh without invoking casper (reducer + rspace only)

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
no ticket

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: rspace-bench/src/test/scala/coop/rchain/rspace/bench/RhoReplayBenchBaseState.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    val randRun = rand
+    assert(createTest(term)(readErrors, runtime.reducer, randRun).unsafeRunSync.isEmpty)
+    (for {
+      setupCheckpoint <- runtime.space.createCheckpoint()
+      _               <- runtime.replaySpace.rigAndReset(setupCheckpoint.root, setupCheckpoint.log)
+      _ = assert(
+        createTest(setupTerm)(readErrors, runtime.replayReducer, randSetup).unsafeRunSync.isEmpty
+      )
+      _ = runTask = createTest(term)(readErrors, runtime.replayReducer, randRun)
```
Comment (lukasz-golebiewski) [inline rspace-bench/src/test/scala/coop/rchain/rspace/bench/RhoReplayBenchBaseState.scala]: I know this code was merely moved, but why does `createTest` need to run twice? (see `assert` above)

File: rspace-bench/src/test/scala/coop/rchain/rspace/bench/RhoBenchBaseState.scala [hunk 2]
```diff
@@ -16,22 +13,29 @@ import coop.rchain.shared.Log
...
+import scala.concurrent.duration._
 
-abstract class WideBenchBaseState {
-  val rhoSetupScriptPath: String = "/rholang/wide-setup.rho"
-  val rhoScriptSource: String    = "/rholang/wide.rho"
+abstract class RhoBenchBaseState {
 
-  implicit val scheduler: Scheduler = Scheduler.fixedPool(name = "wide-1", poolSize = 100)
+  def setupRho: Option[String] = None
+  def testedRho: Option[String]
```
Comment (lukasz-golebiewski) [inline rspace-bench/src/test/scala/coop/rchain/rspace/bench/RhoBenchBaseState.scala]: I can't see this ever being a `None`. Why not simply change it to `String`?

**Comment by lukasz-golebiewski** [comment]
Are those still RSpace benches? I would opt for introducing a `rholangBench` project

File: rspace-bench/src/test/scala/coop/rchain/rspace/bench/LoopContractBench.scala [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  @OutputTimeUnit(TimeUnit.MILLISECONDS)
+  @Fork(value = 1)
+  @Threads(1)
+  @Warmup(iterations = 2)
+  @Measurement(iterations = 5)
+  def loopTwo(bh: Blackhole, state: LoopTwoContractBenchState): Unit = {
+    implicit val scheduler = state.scheduler
+    val result             = state.execute.unsafeRunSync
+    assert(result.isEmpty)
+    bh.consume(processErrors(result))
```
Comment (ArturGajowy) [inline rspace-bench/src/test/scala/coop/rchain/rspace/bench/LoopContractBench.scala]: I think those 4 lines are duplicated in every single `@Benchmark` using RhoBenchBaseState. Could we have them in a single method?

**Review by ArturGajowy — APPROVED** [review]
Neat! I'd love the 4 lines of duplication extracted into a method though :)

File: rspace-bench/src/test/scala/coop/rchain/rspace/bench/RhoReplayBenchBaseState.scala (hunk 1 above)
Comment (dzajkowski) [reply rspace-bench/src/test/scala/coop/rchain/rspace/bench/RhoReplayBenchBaseState.scala]: ~~good point, I'll see what I can do about it~~

Comment (dzajkowski) [reply rspace-bench/src/test/scala/coop/rchain/rspace/bench/RhoReplayBenchBaseState.scala]: the params are different ;)

File: rspace-bench/src/test/scala/coop/rchain/rspace/bench/LoopContractBench.scala (hunk 3 above)
Comment (dzajkowski) [reply rspace-bench/src/test/scala/coop/rchain/rspace/bench/LoopContractBench.scala]: let me try, I think I can make it work

**Comment by dzajkowski** [comment]
bors try

**Pull request merged**